### DISCUSSION TITLE
how can I filtering select results with link properties?

### DISCUSSION THREAD

**Discussion by junhee-yoo** [geldata/gel#4501] [Q&A]
Hi, I'm trying to do a `select` query with `filter by link properties`.

I've tried with:

schema:
```
module default {
    abstract link test_link {
        property tag -> str;
    }

    type Foo {
        property name -> str;
        multi link bars extending test_link -> Bar;
    }

    type Bar {
        property name -> str;
    }
}
```

With the above schema, I could select `Foo` with `Bar` like:
```
select Foo {
  id,
  name,
  bars: {
    id,
    name,
    @tag
  }
}
```
(I referred https://www.edgedb.com/docs/guides/cheatsheet/link_properties#querying)

```json
[
  {
    "id": "09ed064c-4663-11ed-8868-afc54df19e90",
    "name": "foo1",
    "bars": [
      {
        "id": "09ed143e-4663-11ed-8868-1f44ce65412b",
        "name": "bar1",
        "@tag": null
      }
    ]
  },
  {
    "id": "66f1fd52-4663-11ed-8868-43e7063c1191",
    "name": "foo1",
    "bars": [
      {
        "id": "66f20072-4663-11ed-8868-3370ca39b9c5",
        "name": "bar1",
        "@tag": "foo1_bar1_tag"
      }
    ]
  }
]
```

but when I try to filter out bars with link properties, such like:
```
select Foo {
  id,
  name,
  bars: {
    id,
    name,
    @tag
  }
} filter .bars.@tag = 'foo1_bar1_tag'
```

It showed me an error: `EdgeQLSyntaxError: Unexpected '@'` so I removed the '@', `InvalidReferenceError: object type 'default::Bar' has no link or property 'tag'`. I also tried some other ways but still faced errors.

I've searched on issues and discussions on this repo to get an answer but I couldn't find one.

Can I get some advice?

My edgedb server version is:

```bash
$ edgedb-server --version
edgedb-server, version 2.3+d2022092600.gf24c5d924.cv202207260000.r202209260031.tmfqxey3igy2c25lonnxg653ofvwgs3tvpawwo3tv.bofficial.s7ebea42
```

**Comment by niowniow** [comment]
You could assign your tag to a different name and filter using that name. In your case something like this should work
```
select Foo {
  id,
  name,
  bars: {
    id,
    name,
    bar_tag := @tag
  }
} filter .bars.bar_tag = 'foo1_bar1_tag'
```

If that's the best solution, I do not know. I also couldn't find anything about it in the documentation.

**Reply by elprans** [reply]
The error in OP's query is that the correct link property reference syntax is `@tag`, not `.@tag`, so:

```
select Foo {
  id,
  name,
  bars: {
    id,
    name,
  }
} filter .bars@tag = 'foo1_bar1_tag'
```

should work also
